### PR TITLE
Add support for i1 textures

### PR DESCRIPTION
**Pull request by MegaMech**
Changes untested as cannot test atm. Opened draft for now.

### CODE REVIEW

**Review by ethteck — DISMISSED** [review]
The math here isn't right - it's assuming 4 bits per pixel. I recommend writing a small test script so you can confirm the output is correct

**Comment by MegaMech** [comment]
I can successfully export i1 to png now. However,

1) I put black pixels instead of white for ease of visibility. Is that fine?
2) Something weird with
```
iter_image_indexes(
            , , 0.249, 1
```
~~If I set this to 0.25 or 0.125 I get out of range errors. Also tried 0.25, 0.25. If I put 0.0625 the loop doesn't run enough times. That would be the theoretical correct number as this is per bit not per byte or per nibble.~~ (fixed bad math)

In n64graphics is looks like the final size is devised by eight instead of two. Is this needed?

**Comment by ethteck** [comment]
I don't think a graphics extraction library should make any unnecessary transformations on the data, because those transformations would need to be re-applied when building back the blob from a human-viewable format. I would rather this be done in some post-processing step as desired by the project.

if it's 8 bits per pixel, then the size (in pixels) should be calculated by dividing the width * height by 8 I would expect

**Comment by MegaMech** [comment]
Fixed the 0.125 iter thing
Pixels set to white
size devised by eight

**Comment by MegaMech** [comment]
Recommend squash and merge to get rid of the extra commits.

**Comment by MegaMech** [comment]
Pixels now black and white.

alpha is still enabled as I'm not sure how to `img += bytes(p)` I just hard-coded the alpha `bytes((p, 0xFF))` instead.
i4 does `img += bytes(i1, i2)`.

So, my test texture worked flawlessly (mk64 crash_screen_font). However, boot_font is missing data. I think it's an issue with iter bytes_per_x. Theoretically, at 0.125 I shouldn't have to iterate over the bits as it should already iterate enough 8 times per char.

File: n64img/image.py [hunk 1]
```diff
@@ -76,6 +76,37 @@ def size(self) -> int:
 class CI8(Image):
     pass
 
+# I1, a very primitive type where each bit represents one pixel.
+# Generally, only used for debug fonts and rendered using the cpu instead of the rdp.
+class I1(Image):
+    def __init__(self, data, width, height):
+        super().__init__(data, width, height)
+        self.greyscale = True
+        self.alpha = True
```
Comment (ethteck) [inline n64img/image.py]: I think we want this to be false

File: n64img/image.py [hunk 2]
```diff
@@ -76,6 +76,37 @@ def size(self) -> int:
...
+
+            # iterate over bits
+            for j in range(8, 0, -1):
+                # Store the value of each bit as a pixel.
+                p = (b >> (j - 1)) & 0x1
+
+                # Convert active bits to RGB white and inactive to black.
+                p = ceil(0xFF * p)
+
+                img += bytes((p, 0xFF))
```
Comment (ethteck) [inline n64img/image.py]: and then I think we don't need the alpha value here at all and we can just append p

**Comment by MegaMech** [comment]
Thanks!

Will debug more tomorrow (ipl font still not outputting right)

**Comment by MegaMech** [comment]
So, for some reason the iter.iter_image_indexes loop is not running enough times for the ilp3 font. It works fine with mk64's crash_screen_font.
If I put 1, 1, and break the loop if i is greater than self.data that renders a full image.

0.125, 1, seems like it should be right? Unless it should be like 0.5, 0.5?

**Comment by ethteck** [comment]
hmmm I haven't looked at that function very closely. let me see

**Pull request merged**